Automattic/jetpack-onboarding
# Wait for data to finish

While testing, I noticed that I would be asked to wait before closing the panel after completing the walkthrough. At first it seemed to be a firefox-specific issue, but really it's just that the data wasn't finished saving. If I click "Stay on page", the spinner stay on-screen, and blocks my way to try closing the panel again.

This PR adds a new function `afterSavingData`, which lets you pass through a callback to be run once all the data is saved - immediately if nothing is waiting, or by checking every half-second.

This is used on both the Woo step, and before closing the panel. Otherwise there was the (probably small) possibility of users losing the last steps they've just set.

To test:

- On a slower server (fake this by adding `sleep( 3 )` to line 27 in `class.jetpack-onboarding-end-points.php`), run through the onboarding
- Do some configuration, and get to the "Setup your store" step
- Click the button, you should wait a little, while the data finishes saving
- Eventually you should get to the woo step, with all your setup safely saved 😄 

You can also test by not setting up WooCommerce, and instead walking through to the end. Closing the panel should also have you wait for a bit before reloading the page.
